src/main/java/io/cdap/plugin/salesforce/SalesforceBulkUtil.java

```diff
@@ -68,6 +69,7 @@ public final class SalesforceBulkUtil {
   public static JobInfo createJob(BulkConnection bulkConnection, String sObject, OperationEnum operationEnum,
                                   @Nullable String externalIdField,
                                   ConcurrencyMode concurrencyMode, ContentType contentType) throws AsyncApiException {
+    BulkConnectionRetryWrapper bulkConnectionRetryWrapper = new BulkConnectionRetryWrapper(bulkConnection);
```

Comment:
Instead of creating object of RetryWrapper, create it once in getBulkConnection method and rename it and return wrapper instead of bulkConnection. Retry criteria fields also fetch from config instead of hard coding as given now

Reply:
> Retry criteria fields also fetch from config instead of hard coding as given now

Which config ?
There are no fields for retry